First attempt

carolyngreeno opened the pull request:
Finished routes inside app and then commented out and refactored in
routes/truckRoutes.js. Couldn’t get that working.

Commit:
First attempt

Finished routes inside app and then commented out and refactored in
routes/truckRoutes.js. Couldn’t get that working.

jamdon commented on routes/truckRoutes.js:
```diff
@@ -1,3 +1,70 @@
+var router = express.Router();
+
+// '/trucks' routes
+app.route('/trucks')
```
You want to use the router object you just created here instead of app. In app.js you're using app for your application instance. Additionally, since you're mounting the truckRoutes middle-ware on '/trucks' in app.js, the path you use here should be relative to that:

`router.route('/')`

jamdon commented on routes/truckRoutes.js:
```diff
@@ -1,3 +1,70 @@
+
+
+// '/trucks/truckId' routes
+app.route('/trucks/truckId')
```
Same deal here - this should be router, and the path should be relative to /trucks. Additionally, the dynamic portion of the route requires a ':' before the parameter name:

`router.route('/:truckId')`

jamdon commented on routes/truckRoutes.js:
```diff
@@ -1,3 +1,70 @@
+app.route('/trucks/truckId')
+
+	// Retrieve a truck object
+	.post(function (request, response) {
```
You don't want to define a post handler for this route. Using post is for adding a new food truck, and this route already has an id, so it would be either get, put, or delete on an existing truck.

jamdon commented:
Feedback inline. Let me know if you want it graded as is or if you want to submit more updates first.

Commit:
Final edits

Still not working but this is as far as I can get it.

Commit:
removed some var

Moved the following to truckRoutes.js:
- var mongoose
- var db =  mongoose.connect
- var Truck
- app.use

Commit:
argh!

Still can’t get it to work. This is the error I’m getting:

ReferenceError: app is not defined
    at Object.<anonymous>
(/Applications/MAMP/htdocs/jscript.dev/300/week_06/Module_5/routes/truck
Routes.js:3:13)

jamdon commented on routes/truckRoutes.js:
```diff
@@ -1,3 +1,67 @@
-// remember this is a Node module
+var express = require('express');
+var mongoose = require('mongoose');
+var Truck = app.route('../models/truckModel');
```
When you define Truck here, you want to set it to the result of a require call:

`var Truck = require('../models/truckModel');`

You don't need to use app.route anywhere in your application because you'll be using a router instance which you create below on line 5 instead.

jamdon commented on app.js:
```diff
@@ -0,0 +1,17 @@
+var mongoose = require('mongoose');
+
+var bodyParser = require('body-parser');
+var Truck = require('./routes/truckRoutes') (Truck);
```
When you require your truckRoutes module in, assign it to a truckRouter variable instead:

`var truckRouter = require('./routes/truckRoutes');`

You don't want to pass in Truck to a function invocation here, because your truckRoutes isn't returning a function that can be called, it's returning an object.

jamdon commented:
I left a couple more inline comments to address the additional issues you're hitting.

Commit:
final updates

working code

jamdon commented on package.json:
```diff
@@ -0,0 +1,27 @@
+    "body-parser": "^1.12.4",
+    "express": "^4.12.3",
+    "mongoose": "^4.0.3",
+    "nodemon": "^1.3.7",
```
When you install nodemon, I would avoid installing it as a dependency for your project as it's not part of the code you would deploy to production. This is just a tool a developer chooses to use or not locally.

jamdon commented:
:+1:
